jetty-webapp/src/test/java/org/eclipse/jetty/webapp/HugeResourceTest.java

```diff
@@ -0,0 +1,203 @@
+        makeStaticFile(hugeStaticBase.resolve("test-10g.dat"), 10 * GB);
+
+        outputDir = MavenTestingUtils.getTargetTestingPath(HugeResourceTest.class.getSimpleName() + "-huge-static-outputdir");
+        FS.ensureEmpty(outputDir);
+    }
+
+    @AfterAll
+    public static void cleanupHugeStaticFiles()
+    {
+        FS.ensureDeleted(hugeStaticBase);
+        FS.ensureDeleted(outputDir);
+    }
+
+    private static void makeStaticFile(Path staticFile, long size) throws IOException
+    {
+        byte[] buf = new byte[(int)(1 * MB)];
+        Arrays.fill(buf, (byte)'x');
+        ByteBuffer src = ByteBuffer.wrap(buf);
+
+        if (Files.exists(staticFile) && Files.size(staticFile) == size)
+        {
+            // all done, nothing left to do.
+            return;
+        }
+
+        System.err.printf("Creating %,d byte file: %s ...%n", size, staticFile.getFileName());
+        try (SeekableByteChannel channel = Files.newByteChannel(staticFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING))
+        {
+            long remaining = size;
+            while (remaining > 0)
+            {
+                ByteBuffer slice = src.slice();
+                int len = buf.length;
+                if (remaining < Integer.MAX_VALUE)
+                {
+                    len = Math.min(buf.length, (int)remaining);
+                    slice.limit(len);
+                }
+
+                channel.write(slice);
+                remaining -= len;
+            }
+        }
+        System.err.println(" Done");
+    }
+
+    @BeforeEach
+    public void startServer() throws Exception
+    {
+        server = new Server();
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        server.addConnector(connector);
+
+        WebAppContext context = new WebAppContext();
+        context.setContextPath("/");
+        context.setBaseResource(new PathResource(hugeStaticBase));
+
+        HandlerList handlers = new HandlerList();
+        handlers.addHandler(context);
+        handlers.addHandler(new DefaultHandler());
+
+        server.setHandler(handlers);
+        server.start();
+    }
+
+    @AfterEach
+    public void stopServer() throws Exception
+    {
+        server.stop();
+    }
+
+    @Test
+    public void testDownload_1G() throws IOException
+    {
+        download(server.getURI().resolve("/test-1g.dat"), 1 * GB);
+    }
+
+    @Test
+    public void testDownload_4G() throws IOException
+    {
+        download(server.getURI().resolve("/test-4g.dat"), 4 * GB);
+    }
+
+    @Test
+    public void testDownload_10G() throws IOException
+    {
+        download(server.getURI().resolve("/test-10g.dat"), 10 * GB);
+    }
+
+    private void download(URI destUri, long expectedSize) throws IOException
+    {
+        HttpURLConnection http = (HttpURLConnection)destUri.toURL().openConnection();
```

Comment:
Why, oh why??? Use `HttpClient`, so we also know if the client works well with these files.